git2-curl/src/lib.rs

```diff
@@ -75,6 +75,7 @@ pub unsafe fn register(handle: Easy) {
     let handle = Arc::new(Mutex::new(handle));
     let handle2 = handle.clone();
     INIT.call_once(move || {
+        libgit2_sys::init();
```

Comment:
In general we try to auto-initialize libgit2 from the `git2` wrapper crate, so could this call actually be moved into `git2::transport::register`? I think there's a top-level `::init()` function as well in that crate

Reply:
Moved `libgit2_sys::init()` into git2::transport::register.  4b4a1be

I played with moving it into places like: https://github.com/rust-lang/git2-rs/blob/master/src/lib.rs#L682, but no joy. Also not sure I understand how https://github.com/rust-lang/git2-rs/blob/master/src/lib.rs#L686 ties into:
https://github.com/rust-lang/git2-rs/blob/master/libgit2-sys/lib.rs#L3040
and: https://github.com/rust-lang/git2-rs/blob/master/libgit2-sys/lib.rs#L1594